File: JavaScript/7js 2/jumpingNumber.js
/*
|--------------------------------------------------------------------------
| Jumping Number (Special Numbers Series #4)
|--------------------------------------------------------------------------
|
*/

function jumpingNumber(n) {
   let arr = n.toString().split("");
   for (i = 0; i < arr.length - 1; i++) {
      if (Math.abs(arr[i] - arr[i + 1]) != 1) {
         return "Not!!";
      }
   }
   return "Jumping!!";
}


function jumpingNumber(n) {
   return [...(n + "")].every((x, i, a) => !i || x == +a[i - 1] + 1 || x == a[i - 1] - 1)
      ? "Jumping!!"
      : "Not!!";
}


function jumpingNumber(n) {
   return n
      .toString()
      .split("")
      .filter((a, b, c) => Math.abs(a - c[b + 1]) != 1).length > 1
      ? "Not!!"
      : "Jumping!!";
}


console.log(jumpingNumber(32), "Jumping!!");
console.log(jumpingNumber(79), "Not!!");
console.log(jumpingNumber(98), "Jumping!!");
console.log(jumpingNumber(98789876), "Jumping!!");

File: Python/7py 2/jumping_number.py
"""
|--------------------------------------------------------------------------
| Jumping Number (Special Numbers Series #4)
|--------------------------------------------------------------------------
|
"""

def jumping_number(number):
   arr = list(map(lambda i: int(i), list(str(number))))
   for i in range(len(arr)-1):
      if abs(arr[i] - arr[i+1])!=1:
         return "Not!!"
   return "Jumping!!"


def jumping_number(number):
   arr = list(map(int, str(number)))
   return ('Not!!', 'Jumping!!')[all(map(lambda a, b: abs(a - b) == 1, arr, arr[1:]))]


def jumping_number(number):
   s = str(number)
   return next(('Not!!' for a,b in zip(s, s[1:]) if abs(int(b)-int(a)) != 1), 'Jumping!!')


print(jumping_number(98), "Jumping!!")
print(jumping_number(79), "Not!!")
print(jumping_number(4343456) , "Jumping!!")